test/graph_rewrites/test_ng_var_update_ng_tensor.cc

```
@@ -0,0 +1,129 @@
+                  .Attr("ngraph_variable_shared_name", "var1")
+                  .Attr("ngraph_graph_id", 1)
+                  .Finalize(node_def()));
+    ASSERT_OK(InitOp());
+  }
+};
+
+TEST_F(NGVarUpdateNGTensorOpTest, KernelTest) {
+  list<string> env_vars{"NGRAPH_TF_NGVARIABLE_BUFFER_SHARING"};
+  const unordered_map<string, string>& env_map = StoreEnv(env_vars);
+  SetEnvVariable("NGRAPH_TF_NGVARIABLE_BUFFER_SHARING", "0");
+
+  // Create a normal TF tensor: input_tf_tensor and assign values
+  // This will be used to assign initial value to the TF tensor
+  // that is a part of the NGraph Var resource
+  Tensor input_tf_tensor(DT_FLOAT, TensorShape({2}));
+  auto input_tf_tensor_flat = input_tf_tensor.flat<float>();
+  for (size_t i = 0; i < input_tf_tensor_flat.size(); i++) {
+    input_tf_tensor_flat.data()[i] = 1.0;
+  }
+
+  // Create an NGraphVar resource object
+  NGraphVar* var = new NGraphVar(DT_FLOAT, TensorShape{2}, "CPU");
+
+  // Assign the ng_tensor with initial value and use it to copy an
+  // initial value to the tf_tensor
+  // After this, both ng_tensor and tf_tensor will hold the same value
+  // i.e. (1.0, 1.0)
+  var->update_ng_tensor(&input_tf_tensor);
+  var->copy_ng_to_tf();
+
+  // Create a normal TF tensor: input_ng_tensor and assign values
+  // This will be used to assign initial value to the NG tensor
+  // that is a part of the NGraph Var resource
+  Tensor input_ng_tensor(DT_FLOAT, TensorShape({2}));
+  auto input_ng_tensor_flat = input_ng_tensor.flat<float>();
+  // Assign a new value to the normal TF tensor: input_tensor
+  for (size_t i = 0; i < input_ng_tensor_flat.size(); i++) {
+    input_ng_tensor_flat.data()[i] = 5.0;
+  }
+
+  // Update the ng_tensor with the new value i.e. (5.0, 5.0)
+  var->update_ng_tensor(&input_ng_tensor);
+  // So now, both ng_tensor and tf_tensor have different values
+  // which is the desired configuration for the test
+
+  // Create NGraphVariableUpdateNGTensor node
+  MakeOp();
```

Does this need an assert?

---

There is an Assert inside MakeOp to make sure the node gets built.

---

Do we need to propagate the error? by doing ASSERT_OK() ?

---

I think the ASSERT_OK() inside the MakeOp should suffice.

---

You want the execution to stop here, that will only happen if you ASSERT_OK on MakeOp()

---

I think @shresthamalik is talking about an arcane property of gtest that it will continue running unless the failure in in the main body of the test (and not in a subfunction)

---

Got it, I will make the required change.